common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -105,7 +109,23 @@ public void fetchBlocks(
           (blockIds1, listener1) -> {
             // Unless this client is closed.
             if (clientFactory != null) {
-              TransportClient client = clientFactory.createClient(host, port);
+              TransportClient client = null;
+              try {
+                client = clientFactory.createClient(host, port);
+              } catch (InterruptedException e) {
+                throw e;
+              } catch (IOException e) {
+                if (!isHostReachable(host, 3000) || !isHostPortConnectable(host, port)) {
```

Comment:
It's also possible that `Host` is unreachable or connectable when network is unstable, right?